### PR TITLE
cleanup after #453

### PR DESCRIPTION
**Pull request by niklasad1**
Fixes https://github.com/paritytech/jsonrpsee/pull/453#discussion_r704268749 and some other chores

### CODE REVIEW

File: proc-macros/src/helpers.rs [hunk 1]
```diff
@@ -170,9 +170,12 @@ pub(crate) fn is_option(ty: &syn::Type) -> bool {
 /// Note that `doc comments` are expanded into `#[doc = "some comment"]`
 /// Thus, if the attribute starts with `doc` => it's regarded as a doc comment.
 pub(crate) fn extract_doc_comments(attrs: &[syn::Attribute]) -> TokenStream2 {
-	let docs = attrs.iter().filter(|attr| attr.path.is_ident("doc")).filter(|attr| match attr.parse_meta() {
-		Ok(syn::Meta::NameValue(meta)) if matches!(&meta.lit, syn::Lit::Str(_)) => true,
-		_ => false,
+	let docs = attrs.iter().filter(|attr| {
```
Comment (niklasad1) [inline proc-macros/src/helpers.rs:173]: a little bit better @maciejhirsz :)

**Review by jsdw — APPROVED** [review]
LGTM :)

File: ws-server/src/tests.rs [hunk 2]
```diff
@@ -122,22 +122,22 @@ async fn server_with_context() -> SocketAddr {
 
 	rpc_module
 		.register_method("should_err", |_p, ctx| {
-			let _ = ctx.err().map_err(|e| CallError::Failed(e.into()))?;
+			let _ = ctx.ok().map_err(CallError::Failed)?;
```
Comment (maciejhirsz) [inline ws-server/src/tests.rs:125]: Is the swap from `err` to `ok` here intentional?

**Review by maciejhirsz — APPROVED** [review]
Looks clean! One question :)

File: ws-server/src/tests.rs (hunk 2 above)
Comment (niklasad1) [inline ws-server/src/tests.rs:125]:
```suggestion
			let _ = ctx.err().map_err(CallError::Failed)?;
```

File: http-server/src/tests.rs [hunk 3]
```diff
@@ -66,21 +66,21 @@ async fn server_with_handles() -> (SocketAddr, JoinHandle<Result<(), Error>>, St
 	module.register_method("notif", |_, _| Ok("")).unwrap();
 	module
 		.register_method("should_err", |_, ctx| {
-			let _ = ctx.err().map_err(|e| CallError::Failed(e.into()))?;
+			let _ = ctx.ok().map_err(CallError::Failed)?;
```
Comment (niklasad1) [inline http-server/src/tests.rs:69]:
```suggestion
			let _ = ctx.err().map_err(CallError::Failed)?;
```

File: ws-server/src/tests.rs (hunk 2 above)
Comment (niklasad1) [reply ws-server/src/tests.rs]: copy-past nit but those are fixed now, good catch

**Pull request merged**